lib/service/service.go

```diff
@@ -346,7 +348,25 @@ func (process *TeleportProcess) initAuthService(authority auth.Authority) error
 
 	// Heart beat auth server presence, this is not the best place for this
 	// logic, consolidate it into auth package later
-	var authClient *auth.TunClient
+	storage := utils.NewFileAddrStorage(
```

Comment:
it does what you need. no reason for code duplication. tun clients are messy and leaky, there's a function which creates one for you (and checks for connectivity to boot!)